Error logging in a second with npm adduser

<!--
First of all, please read this section

https://github.com/verdaccio/verdaccio/blob/master/CONTRIBUTING.md#reporting-a-bug

Some advices before file an issue
  * Give a brief explanation of the issue, suggestion or feature to request.
  * If the issue is a question, provide as much information you have available.
  * How can I do in order to reproduce it? What environment?
  * Define which version the issue happens and whether previous version the behaviour is correct.
  * Provide your config file is really helpful. Please be aware to hide sensisive data (passwords, server IP, etc) before post.
-->

#### My reason: Trying to login using an existing user account generates errors on npm
#### Steps to reproduce:
When running Verdaccio on localhost port 4873:
 - Run ```npm adduser --registry=localhost:4873``` to create a user account with a password
 - Run ```npm adduser --registry=localhost:4873``` to create a second user account with a password
 - Run ```npm adduser --registry=localhost:4873``` with the first user account and password
 - npm will inform the user of the following error
  ```npm ERR! Cannot create property 'username' on boolean 'true'```
The npm error reported seems to be because npm is expecting a json object in the body but the response is simply 'true' with a status code 200. This response differs from what the htpasswd code from v2.7.4 of Verdaccio returns in this case which is a status code 409 with an error that the user already exists. npm will also not change the user to the first user, which can be verified by running ```npm whoami --registry=http://localhost:4873```.
#### App Version: 
- node v8.9.0
- verdaccio-htpasswd v0.1.4
- verdaccio v3.0.0-beta.4
- npm v5.7.1
#### Config file: 
```
storage: ~/.local/share/verdaccio/storage
auth:
  htpasswd:
    file: ./htpasswd
    max_users: 1000
uplinks:
  npmjs:
    url: https://registry.npmjs.org/
packages:
  '@*/*':
    access: $authenticated
    publish: $authenticated
    proxy: npmjs
  '**':
    access: $authenticated
    publish: $authenticated
    proxy: npmjs
```
#### Additional information:

 - Verdaccio terminal output
```
 warn --- config file  - /Users/tapps/.config/verdaccio/config.yaml
 warn --- Plugin successfully loaded: htpasswd
 warn --- http address - http://localhost:4873/ - verdaccio/3.0.0-beta.4
 http <-- 404, user: undefined(127.0.0.1), req: 'POST /-/v1/login', bytes: 34/150
 http <-- 201, user: test(127.0.0.1), req: 'PUT /-/user/org.couchdb.user:test', bytes: 122/73
 http <-- 404, user: undefined(127.0.0.1), req: 'POST /-/v1/login', bytes: 34/150
 http <-- 200, user: undefined(127.0.0.1), req: 'PUT /-/user/org.couchdb.user:test', bytes: 122/4
 http <-- 200, user: undefined(127.0.0.1), req: 'PUT /-/user/org.couchdb.user:test', bytes: 146/4
```

 - Which (Windows, OS X/macOS, or Linux) environment are you running verdaccio?: OS X

#### Additional verbose log:
```
  express:router dispatching POST /-/v1/login +12s
  express:router query  : /-/v1/login +1ms
  express:router expressInit  : /-/v1/login +0ms
  express:router corsMiddleware  : /-/v1/login +1ms
  express:router log  : /-/v1/login +0ms
  express:router errorReportingMiddleware  : /-/v1/login +1ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router compression  : /-/v1/login +0ms
  express:router router  : /-/v1/login +0ms
  express:router dispatching POST /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router jsonParser  : /-/v1/login +1ms
  express:router <anonymous>  : /-/v1/login +16ms
  express:router encodeScopePackage  : /-/v1/login +0ms
  express:router router  : /-/v1/login +1ms
  express:router dispatching POST /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router securityIframe  : /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router final  : /-/v1/login +0ms
 http <-- 404, user: undefined(127.0.0.1), req: 'POST /-/v1/login', bytes: 34/150
  express:router dispatching PUT /-/user/org.couchdb.user:test +12s
  express:router query  : /-/user/org.couchdb.user:test +0ms
  express:router expressInit  : /-/user/org.couchdb.user:test +0ms
  express:router corsMiddleware  : /-/user/org.couchdb.user:test +0ms
  express:router log  : /-/user/org.couchdb.user:test +0ms
  express:router errorReportingMiddleware  : /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router compression  : /-/user/org.couchdb.user:test +0ms
  express:router router  : /-/user/org.couchdb.user:test +1ms
  express:router dispatching PUT /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router jsonParser  : /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router encodeScopePackage  : /-/user/org.couchdb.user:test +0ms
 http <-- 201, user: test(127.0.0.1), req: 'PUT /-/user/org.couchdb.user:test', bytes: 126/73
  express:router router  : /-/user/org.couchdb.user:test +21ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +1ms
  express:router final  : /-/user/org.couchdb.user:test +0ms
  express:router dispatching POST /-/v1/login +3s
  express:router query  : /-/v1/login +0ms
  express:router expressInit  : /-/v1/login +0ms
  express:router corsMiddleware  : /-/v1/login +0ms
  express:router log  : /-/v1/login +0ms
  express:router errorReportingMiddleware  : /-/v1/login +1ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router compression  : /-/v1/login +0ms
  express:router router  : /-/v1/login +0ms
  express:router dispatching POST /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router jsonParser  : /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router encodeScopePackage  : /-/v1/login +0ms
  express:router router  : /-/v1/login +0ms
  express:router dispatching POST /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router securityIframe  : /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router final  : /-/v1/login +0ms
 http <-- 404, user: undefined(127.0.0.1), req: 'POST /-/v1/login', bytes: 34/150
  express:router dispatching PUT /-/user/org.couchdb.user:test2 +12s
  express:router query  : /-/user/org.couchdb.user:test2 +0ms
  express:router expressInit  : /-/user/org.couchdb.user:test2 +0ms
  express:router corsMiddleware  : /-/user/org.couchdb.user:test2 +1ms
  express:router log  : /-/user/org.couchdb.user:test2 +0ms
  express:router errorReportingMiddleware  : /-/user/org.couchdb.user:test2 +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test2 +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test2 +0ms
  express:router compression  : /-/user/org.couchdb.user:test2 +0ms
  express:router router  : /-/user/org.couchdb.user:test2 +0ms
  express:router dispatching PUT /-/user/org.couchdb.user:test2 +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test2 +0ms
  express:router jsonParser  : /-/user/org.couchdb.user:test2 +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test2 +0ms
  express:router encodeScopePackage  : /-/user/org.couchdb.user:test2 +0ms
 http <-- 201, user: test2(127.0.0.1), req: 'PUT /-/user/org.couchdb.user:test2', bytes: 128/74
  express:router router  : /-/user/org.couchdb.user:test2 +13ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test2 +1ms
  express:router final  : /-/user/org.couchdb.user:test2 +0ms
 http <-- 404, user: undefined(127.0.0.1), req: 'POST /-/v1/login', bytes: 34/150
  express:router dispatching POST /-/v1/login +3s
  express:router query  : /-/v1/login +0ms
  express:router expressInit  : /-/v1/login +0ms
  express:router corsMiddleware  : /-/v1/login +0ms
  express:router log  : /-/v1/login +0ms
  express:router errorReportingMiddleware  : /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router compression  : /-/v1/login +0ms
  express:router router  : /-/v1/login +0ms
  express:router dispatching POST /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router jsonParser  : /-/v1/login +1ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router encodeScopePackage  : /-/v1/login +0ms
  express:router router  : /-/v1/login +0ms
  express:router dispatching POST /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router securityIframe  : /-/v1/login +0ms
  express:router <anonymous>  : /-/v1/login +0ms
  express:router final  : /-/v1/login +0ms
  express:router dispatching PUT /-/user/org.couchdb.user:test +7s
  express:router query  : /-/user/org.couchdb.user:test +0ms
  express:router expressInit  : /-/user/org.couchdb.user:test +0ms
  express:router corsMiddleware  : /-/user/org.couchdb.user:test +0ms
  express:router log  : /-/user/org.couchdb.user:test +0ms
  express:router errorReportingMiddleware  : /-/user/org.couchdb.user:test +1ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router compression  : /-/user/org.couchdb.user:test +0ms
  express:router router  : /-/user/org.couchdb.user:test +0ms
  express:router dispatching PUT /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router jsonParser  : /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router encodeScopePackage  : /-/user/org.couchdb.user:test +0ms
  express:router router  : /-/user/org.couchdb.user:test +2ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router final  : /-/user/org.couchdb.user:test +0ms
 http <-- 200, user: undefined(127.0.0.1), req: 'PUT /-/user/org.couchdb.user:test', bytes: 126/4
  express:router dispatching PUT /-/user/org.couchdb.user:test +4ms
  express:router query  : /-/user/org.couchdb.user:test +0ms
  express:router expressInit  : /-/user/org.couchdb.user:test +0ms
  express:router corsMiddleware  : /-/user/org.couchdb.user:test +0ms
  express:router log  : /-/user/org.couchdb.user:test +0ms
  express:router errorReportingMiddleware  : /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router compression  : /-/user/org.couchdb.user:test +0ms
  express:router router  : /-/user/org.couchdb.user:test +0ms
  express:router dispatching PUT /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router jsonParser  : /-/user/org.couchdb.user:test +0ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +1ms
  express:router encodeScopePackage  : /-/user/org.couchdb.user:test +0ms
  express:router router  : /-/user/org.couchdb.user:test +3ms
  express:router <anonymous>  : /-/user/org.couchdb.user:test +0ms
  express:router final  : /-/user/org.couchdb.user:test +0ms
 http <-- 200, user: undefined(127.0.0.1), req: 'PUT /-/user/org.couchdb.user:test', bytes: 150/4
```

--- a/src/___tests___/utils.test.js
+++ b/src/___tests___/utils.test.js
@@ -171,10 +171,11 @@ describe('sanityCheck', () => {
     expect(input.message).toEqual('username and password is required');
     expect(input.status).toEqual(400);
   });
-  it('should successfully authenicate the user', () => {
+  it('should successfully authenticate the user', () => {
     const verifyFn = jest.fn(() => true);
     const input = sanityCheck('test', users.test, verifyFn, users, 2);
-    expect(input).toBeTruthy();
-    expect(verifyFn).toHaveBeenCalled();
+    expect(input.status).toEqual(409);
+    expect(input.message).toEqual('username is already registered');
+    expect(verifyFn).toHaveBeenCalledTimes(1);
   });
 });

--- a/src/___tests___/utils.test.js
+++ b/src/___tests___/utils.test.js
@@ -171,7 +171,7 @@ describe('sanityCheck', () => {
     expect(input.message).toEqual('username and password is required');
     expect(input.status).toEqual(400);
   });
-  it('should successfully authenticate the user', () => {
+  it('should throw error for existing username and password', () => {
     const verifyFn = jest.fn(() => true);
     const input = sanityCheck('test', users.test, verifyFn, users, 2);
     expect(input.status).toEqual(409);

--- a/src/utils.js
+++ b/src/utils.js
@@ -132,7 +132,10 @@ export function sanityCheck(
   if (hash) {
     const auth = verifyFn(password, users[user]);
     if (auth) {
-      return true;
+      err = Error('username is already registered');
+      // $FlowFixMe
+      err.status = 409;
+      return err;
     }
     err = Error('unauthorized access');
     // $FlowFixMe

